src/main/scala/sparkz/core/network/DeliveryTracker.scala

```diff
@@ -250,4 +261,18 @@ class DeliveryTracker(system: ActorSystem,
         log.warn("Unexpected error", e)
         Failure(e)
     }
+
+  private def updateProcessingTime(startTime: Long): Unit = {
+    if (slowModeFeatureFlag) {
+      val elapsedMs: Long = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime)
+      averageProcessingTimeMs = (averageProcessingTimeMs * 0.9).toLong + (elapsedMs * 0.1).toLong
+      if (averageProcessingTimeMs > slowModeThresholdMs && !slowMode) {
+        slowMode = true
+        logger.warn("Slow mode enabled on P2P layer due to high load. Transactions won't be requested or broadcasted.")
+      } else if (averageProcessingTimeMs < slowModeThresholdMs && slowMode) {
+        slowMode = false
+        logger.warn("Slow mode disabled on P2P layer. Transactions will be requested or broadcasted.")
+      }
+    }
+  }
```

Comment:
some things to consider on this formula:
1- we are considering both blocks and transactions modifier in the same way, but blocks may require higher apply time specially if very full of smart contracts txs (but it is also true that in this case  number of tx modifiers vs number of block modifiers will be high, so maybe the average is ok anyway - did you made some thoughts on this already?)
2- consider this corner case: all the nodes enters in "slowmode" and so any new tx will be broadcasted by anyone. don't we risk the node will never go out to the slowmode? do you think the block modifiers will be enough to "lower" the average in an acceptable time?
3- let's test also that  after bootstrapping a new node and making a full synch it will work correctly (in this case I assume a lot of block modifiers and very few tx modifiers during the sync)

Reply:
discussed in Slack, decided to go with a mix of previous and current approach, so that a node keep on requesting some amount of transactions from the network even under the load. This will help with returning from a slow mode state to a regular work state much faster.

As for the tests, I've tested rate limiting with a bunch of tests in STF that generate 2-3-4k blocks